vllm/attention/backends/flash_attn.py

```diff
@@ -300,8 +301,60 @@ def decode_metadata(self) -> Optional["FlashAttentionMetadata"]:
             block_tables=self.block_tables[self.num_prefills:],
             use_cuda_graph=self.use_cuda_graph,
         )
+
         return self._cached_decode_metadata
 
+    # TODO (varun) : Try using decode_metadata here. We hit some asserts in
+    # advance_step - but that seems resolvable.
```

Comment:
Can you elaborate more on this? Specifically how it can be solved and what's the plan?